chain/client/src/view_client.rs

```diff
@@ -445,16 +467,19 @@ impl ViewClientActor {
             .chain
             .get_block_header(&execution_outcome.transaction_outcome.block_hash)?])
         {
-            return if is_execution_finished {
+            return if executed_ignoring_refunds {
                 Ok(TxExecutionStatus::ExecutedOptimistic)
             } else {
                 Ok(TxExecutionStatus::Included)
             };
         }
 
-        if !is_execution_finished {
+        if !executed_ignoring_refunds {
             return Ok(TxExecutionStatus::IncludedFinal);
         }
+        if !executed_including_refunds {
+            return Ok(TxExecutionStatus::Executed);
+        }
```

Comment:
It's confusing to see logic like `if !executed return "Executed"` :)
I wouldn't mind comment like "if we are just waiting for refunds, return "Executed""